File: kirun-js/src/engine/runtime/expression/operators/binary/LogicalNullishCoalescingOperator.ts
import { isNullValue } from '../../../../util/NullCheck';
import { BinaryOperator } from './BinaryOperator';

export class LogicalNullishCoalescingOperator extends BinaryOperator {
    public apply(t: any, u: any): any {
        return isNullValue(t) ? u : t;
    }
}

File: kirun-py/src/kirun_py/runtime/expression/operators/binary/logical_nullish_coalescing_operator.py
from __future__ import annotations

from typing import Any

from kirun_py.util.null_check import is_null_value
from kirun_py.runtime.expression.operators.binary.binary_operator import BinaryOperator


class LogicalNullishCoalescingOperator(BinaryOperator):

    def apply(self, t: Any, u: Any) -> Any:
        return u if is_null_value(t) else t
